### PR TITLE
feat(rebuild index) - add "app:rebuild_elastic_index" command

### PR DESCRIPTION
**Pull request by amagdas**
Depends on https://github.com/petrjasek/eve-elastic/pull/27

### CODE REVIEW

**Comment by amagdas** [comment]
@superdesk/developers

**Comment by ioanpocol** [comment]
:+1:

File: superdesk/commands.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class RebuildElasticIndex(superdesk.Command):
+    def run(self):
+        index_name = superdesk.app.config['ELASTICSEARCH_INDEX']
+        try:
+            es = get_es(superdesk.app.config['ELASTICSEARCH_URL'])
+            clone_name = index_name + '-' + get_random_string()
+            print('Starting index rebuilding for index: ' + index_name)
+            reindex(es, index_name, clone_name)
```
Comment (petrjasek) [inline superdesk/commands.py]: shouldn't it put the mapping before reindexing into new index?

Comment (amagdas) [inline superdesk/commands.py]: I don't think so, I thought that the new mapping is already there when creating the new index.
Is there any example on how this is actually done?

Comment (petrjasek) [inline superdesk/commands.py]: there is [put_mapping](https://github.com/petrjasek/eve-elastic/blob/master/eve_elastic/elastic.py#L168) in `eve-elastic`

Comment (amagdas) [inline superdesk/commands.py]: Hmm, probably updating the mapping for the current index will have weird effects.
What I think will work is the following flow:
- create a new index 
- put new mapping on it
- reindex the data from the old index on the new index

Sounds good?

Comment (petrjasek) [inline superdesk/commands.py]: yep exactly. it's not possible to put mapping once there is some data, it will fail

**Comment by amagdas** [comment]
Updated, did some changes in eve-elastic as well: https://github.com/petrjasek/eve-elastic/pull/27

**Comment by petrjasek** [comment]
:+1:

**Pull request merged**